sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryStrategy.java

```diff
@@ -44,4 +45,15 @@ default boolean shouldRetry(HttpResponse httpResponse) {
             && code != HttpURLConnection.HTTP_NOT_IMPLEMENTED
             && code != HttpURLConnection.HTTP_VERSION));
     }
+
+    /**
+     * This method is consulted to determine if a retry attempt should be made for the given {@link Throwable}
+     * propagated when the request failed to send.
+     *
+     * @param throwable The {@link Throwable} thrown during the previous attempt.
+     * @return Whether a retry should be attempted.
+     */
+    default boolean shouldRetryException(Throwable throwable) {
+        return true;
```

Comment:
Should this always return `true`? Why is this a separate method?

Reply:
This is a separate method as it will only be called on the `onErrorResume` handler. Beforehand, it would retry all exceptions, so this is to maintain previous functionality.